flink-end-to-end-tests/test-scripts/common_ssl.sh

```diff
@@ -0,0 +1,98 @@
+    fi
+    mkdir -p "${ssl_dir}"
+
+    SANSTRING="dns:${NODENAME}"
+    for NODEIP in $(get_node_ip) ; do
+        SANSTRING="${SANSTRING},ip:${NODEIP}"
+    done
+
+    echo "Using SAN ${SANSTRING}"
+
+    # create certificates
+    keytool -genkeypair -alias ca -keystore "${ssl_dir}/ca.keystore" -dname "CN=Sample CA" -storepass ${password} -keypass ${password} -keyalg RSA -ext bc=ca:true -storetype PKCS12
+    keytool -keystore "${ssl_dir}/ca.keystore" -storepass ${password} -alias ca -exportcert > "${ssl_dir}/ca.cer"
+    keytool -importcert -keystore "${ssl_dir}/ca.truststore" -alias ca -storepass ${password} -noprompt -file "${ssl_dir}/ca.cer"
+
+    keytool -genkeypair -alias node -keystore "${ssl_dir}/node.keystore" -dname "CN=${NODENAME}" -ext SAN=${SANSTRING} -storepass ${password} -keypass ${password} -keyalg RSA -storetype PKCS12
+    keytool -certreq -keystore "${ssl_dir}/node.keystore" -storepass ${password} -alias node -file "${ssl_dir}/node.csr"
+    keytool -gencert -keystore "${ssl_dir}/ca.keystore" -storepass ${password} -alias ca -ext SAN=${SANSTRING} -infile "${ssl_dir}/node.csr" -outfile "${ssl_dir}/node.cer"
+    keytool -importcert -keystore "${ssl_dir}/node.keystore" -storepass ${password} -file "${ssl_dir}/ca.cer" -alias ca -noprompt
+    keytool -importcert -keystore "${ssl_dir}/node.keystore" -storepass ${password} -file "${ssl_dir}/node.cer" -alias node -noprompt
+
+    openssl pkcs12 -passin pass:${password} -in "${ssl_dir}/node.keystore" -passout pass: -out "${ssl_dir}/node.pem" -nodes
+
+    # adapt config
+    # (here we rely on security.ssl.enabled enabling SSL for all components and internal as well as
+    # external communication channels)
+    set_conf security.ssl.${type}.enabled true
+    set_conf security.ssl.${type}.keystore ${ssl_dir}/node.keystore
+    set_conf security.ssl.${type}.keystore-password ${password}
+    set_conf security.ssl.${type}.key-password ${password}
+    set_conf security.ssl.${type}.truststore ${ssl_dir}/ca.truststore
+    set_conf security.ssl.${type}.truststore-password ${password}
+}
+
+function set_conf_internal_ssl {
+    _set_conf_ssl_helper "internal"
+}
+
+function set_conf_mutual_rest_ssl {
+    local auth="${1:-server}" # only 'server' or 'mutual'
+    local mutual="false"
+    local ssl_dir="${TEST_DATA_DIR}/ssl/rest"
+    if [ "${auth}" == "mutual" ]; then
+        CURL_SSL_ARGS="${CURL_SSL_ARGS} --cert ${ssl_dir}/node.pem"
+        mutual="true";
+    fi
+    echo "Mutual ssl auth: ${mutual}"
+    set_conf security.ssl.rest.authentication-enabled ${mutual}
+}
+
+function set_conf_rest_ssl {
+    local auth="${1:-server}" # only 'server' or 'mutual'
+    local ssl_dir="${TEST_DATA_DIR}/ssl/rest"
+    _set_conf_ssl_helper "rest"
+    set_conf_mutual_rest_ssl ${auth}
+    REST_PROTOCOL="https"
+    CURL_SSL_ARGS="${CURL_SSL_ARGS} --cacert ${ssl_dir}/node.pem"
+}
+
+function set_conf_ssl {
+    local auth="${1:-server}" # only 'server' or 'mutual'
```

Comment:
is any test actually using the `server` setting? (if not then I'd say we should only modify 1 test to use mutual.)

Reply:
I will modify batch all around to use only server version